### PR TITLE
Standardise gem types in frontend

### PR DESCRIPTION
**Pull request by sebastiantoh**
- Replace all testing 'gem' implementations with actual gem type

### CODE REVIEW

File: backend/internal/core/domain/gems/gem.go [hunk 1]
```diff
@@ -4,6 +4,17 @@ import (
 	"time"
 )
 
+type GemColor int64
+
+const (
+	Purple GemColor = 0
```
Comment (ianyong) [inline backend/internal/core/domain/gems/gem.go:10]: As a side note, `iota` is a possible alternative, though I think it makes more sense to list out each value given that they need to map to the same values in the frontend.

File: frontend/src/providers/GemProvider.vue [hunk 2]
```diff
@@ -57,7 +81,7 @@ export default defineComponent({
         if (err instanceof Error) {
           return Promise.reject(err);
         }
-        return Promise.reject(new Error("Unknown error type"));
+        return Promise.reject(new Error((err as any).message));
```
Comment (ianyong) [inline frontend/src/providers/GemProvider.vue]: Why not just `Promise.reject(err)`?

Comment (sebastiantoh) [reply frontend/src/providers/GemProvider.vue]: right lol. i think i had some typescript issues which was why I had the above. But Promise.reject(err) works great!

**Review by ianyong — APPROVED** [review]
LGTM!

**Pull request merged**